### DISCUSSION TITLE
Provide topology diagrams for Windows and Linux server monitoring

### DISCUSSION THREAD

**Discussion by changlongmao** [apache/skywalking#11174] [Q&A]
We have successfully integrated monitoring from Windows and Linux servers into Skywalking using OpenTelemetry. We are able to view service information and line charts. I'm interested in learning how to add a server topology diagram, similar to the one in the General Service section. I've attempted to add metrics to a new topology diagram, but the displayed data is not being found.

**Comment by changlongmao** [comment]
<img width="413" alt="image" src="https://github.com/apache/skywalking/assets/53187265/c4154ba9-dc91-449c-9703-07329b4bda1c">
In addition, the topology diagram of the service shows that the condition of unhealthy nodes only provides the "&&" scheme, and we need the function of "||"

**Reply by wu-sheng** [reply]
With the enhancement of MQE, we may have an alternative solution, but right now, no plan to support that.

**Comment by wu-sheng** [comment]
> I'm interested in learning how to add a server topology diagram, similar to the one in the General Service section

It is not a general service. It is a Windows layer service. You could have topo widget on that page by yourself.
BUT, the key question is, where do you have the data? Where do you read the topology? OS level topology is IP based, do you have that traffic data?

**Reply by changlongmao** [reply]
The topology edges are derived from TCP communication between services, with topology nodes representing service endpoints. Data collection from Linux servers is facilitated by OpenTelemetry, which provides network bandwidth metrics. However, it lacks the association context. I've seen that you folks are able to provide topology diagrams for Kubernetes pods using eBPF. Is there a similar solution available for Windows or Linux servers? If not, I would be interested in displaying only the topology nodes without depicting the relational edges. Can this be configured and implemented independently?

**Reply by wu-sheng** [reply]
In theory, you could. If you are asking whether maintainers are going to add this, No plan for that. 
SkyWalking community would not add features based on requests, we only work with contributors to add. How to add, and whether who would add this are your responsibility.
